libs/gltfio/src/Animator.cpp

```diff
@@ -434,7 +437,7 @@ void AnimatorImpl::applyAnimation(const Channel& channel, float t, size_t prevIn
     // Perform the interpolation. This is a simple but inefficient implementation; Filament
     // stores transforms as mat4's but glTF animation is based on TRS (translation rotation
     // scale).
-    mat4f xform = transformManager->getTransform(node);
+    mat4f xform = channel.restTransform;
```

Comment:
I'm not sure if this is correct. This only works if a `Channel` maps to *one* Entity. But for example if 2 `Channel` can animate the same Entity (e.g. one animates the scale and one the rotation), this doesn't work.

On the other hand, if there is a one to one relation between `Channel` and `Entity`, then this code a way more complex and costly than it needs. We could instead store a union of T, R or S. We wouldn't need to do the `decomposeMatrix` (i.e. with this code we don't need it, since it's a constant).

So imho, either this fix is incorrect or the code is more complex than it needs.

Reply:
Yeah, you're right. A `Channel` can affect more than one Entity. I'll have to think of a more robust solution.